src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = /[.\-/]/;
+  const newDate = [];
+  const objectDate = {};
+  const splitDate = date.split(oldSeparator);
+  const isPunctuation = (str) => /^[\p{P}\p{S}]+$/u.test(str);
+  const punct = toFormat.filter(isPunctuation).join('');
+
+  for (let i = 0; i < splitDate.length; i++) {
+    objectDate[fromFormat[i]] = splitDate[i];
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (objectDate['YY'] < 30) {
+      objectDate['YYYY'] = '20' + objectDate['YY'];
+    } else {
+      objectDate['YYYY'] = '19' + objectDate['YY'];
+    }
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    objectDate['YY'] = objectDate['YYYY'].slice(-2);
+  }
+
+  toFormat.forEach((value) => {
+    newDate.push(objectDate[value]);
+  });
+
+  return newDate.join(punct).slice(0, -1);
```

Comment:
The `slice(0, -1)` at the end of the `join` operation is not needed. Since you will have the correct separator from the `toFormat` array, you can join the `newDate` array with it without slicing off the last character.